server/src/main/java/org/tctalent/server/service/db/impl/LinkPreviewServiceImpl.java

```
@@ -0,0 +1,282 @@
+import org.tctalent.server.exception.EntityReferencedException;
+import org.tctalent.server.exception.InvalidRequestException;
+import org.tctalent.server.logging.LogBuilder;
+import org.tctalent.server.model.db.LinkPreview;
+import org.tctalent.server.repository.db.LinkPreviewRepository;
+import org.tctalent.server.service.db.LinkPreviewService;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class LinkPreviewServiceImpl implements LinkPreviewService {
+
+  private final LinkPreviewRepository linkPreviewRepository;
+
+  @Override
+  public boolean deleteLinkPreview(long linkPreviewId)
+      throws EntityReferencedException, InvalidRequestException {
+    linkPreviewRepository.deleteById(linkPreviewId);
+    return true;
+  }
+
+  @Override
+  public @Nullable LinkPreview buildLinkPreview(String url) throws IOException {
+    LinkPreview linkPreview = new LinkPreview();
+    linkPreview.setUrl(url);
+
+    try {
+      Document doc = Jsoup.connect(url).get();
+
+      // We don't want to return a link preview without all the three following values.
+      String description = getDescription(doc);
+      if (description != null) {
+        linkPreview.setDescription(description);
+      } else return null;
+
+      String title = getTitle(doc);
+      if (title != null) {
+        linkPreview.setTitle(title);
+      } else return null;
+
+      String domain = getDomain(doc, url);
+      if (domain != null) {
+        linkPreview.setDomain(domain);
+      } else return null;
+
+      // These are not essential and often may not be available.
+      linkPreview.setImageUrl(getImageUrl(doc));
+      linkPreview.setFaviconUrl(getFaviconUrl(doc));
+
+      return linkPreview;
+
+    } catch (HttpStatusException e) {
+      LogBuilder.builder(log)
+          .action("BuildLinkPreview: Jsoup.connect(url).get()")
+          .message("HTTP error fetching URL " + "(" + url + "): " + e.getStatusCode())
+          .logInfo();
+
+      return null;
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception has been ignored. The catch-block hides information about the stack trace. Not logging or forwarding the stack trace of an exception can mask unexpected errors and complicated debugging.

**Fix**: Use a catch block to handle the exception and write a log message declaring that the exception happened. 
[Learn more.](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)